# Analisi dataset studenti italiani che frequentano l'ora di religione

Questa è un'analisi del dataset ottenuto da onData e UAAR attraverso una richiesta FOIA al Ministero dell'Istruzione, dell'Università e della Ricerca.

## Import di Pandas e dei file da analizzare

In [1]:
import pandas as pd
pd.options.mode.chained_assignment = None  # default='warn'
pd.options.display.float_format = '{:.2f}'.format #display more readable float numbers

In [2]:
# reading every page of the Excel file as a single dataframe

data_2019 = pd.read_excel('religione-cattolica.xlsx', sheet_name=0)
data_2020 = pd.read_excel('religione-cattolica.xlsx', sheet_name=1)
data_2021 = pd.read_excel('religione-cattolica.xlsx', sheet_name=2)

In [3]:
# the index is set to the 'Codice Scuola' column, which contains an unique identifier of the single school

data_2019 = data_2019.set_index('Codice Scuola')
data_2020 = data_2020.set_index('Codice Scuola')
data_2021 = data_2021.set_index('Codice Scuola')

In [4]:
data_2019.info()

<class 'pandas.core.frame.DataFrame'>
Index: 53244 entries, AG1A001006 to VVTL01101X
Data columns (total 1 columns):
 #   Column                                                              Non-Null Count  Dtype 
---  ------                                                              --------------  ----- 
 0   Numero di alunni che frequentano l'ora di religione - a.s. 2018/19  53244 non-null  object
dtypes: object(1)
memory usage: 831.9+ KB


In [5]:
data_2020.info()

<class 'pandas.core.frame.DataFrame'>
Index: 52973 entries, AG1A001006 to VVTL01101X
Data columns (total 1 columns):
 #   Column                                                              Non-Null Count  Dtype 
---  ------                                                              --------------  ----- 
 0   Numero di alunni che frequentano l'ora di religione - a.s. 2019/20  52973 non-null  object
dtypes: object(1)
memory usage: 827.7+ KB


In [6]:
data_2021.info()

<class 'pandas.core.frame.DataFrame'>
Index: 52688 entries, AG1A001006 to VVTN00201L
Data columns (total 1 columns):
 #   Column                                                              Non-Null Count  Dtype 
---  ------                                                              --------------  ----- 
 0   Numero di alunni che frequentano l'ora di religione - a.s. 2020/21  52688 non-null  object
dtypes: object(1)
memory usage: 823.2+ KB


Possiamo già notare come i tre dataset non hanno un numero costante di valori per ogni anno. Possiamo aspettarci che ci siano scuole che compaiono in alcuni anni e su altri anni no. Potremmo eventualmente decidere di riempire i dati con la media degli anni presenti. In questo caso si è preferito lavorare sui numeri che erano presenti, senza fare inferenze di alcun tipo.

## Import dell'anagrafica delle scuole statali e paritarie

In [7]:
# reading state schools
statali_2021 = pd.read_csv("SCUANAGRAFESTAT20202120210831.csv")
statali_2020 = pd.read_csv("SCUANAGRAFESTAT20192020200831.csv")
statali_2019 = pd.read_csv("SCUANAGRAFESTAT20181920190831.csv")

In [8]:
# the index is set to the 'CODICESCUOLA' column, so it will match the column from the data
statali_2019 = statali_2019.set_index('CODICESCUOLA')
statali_2020 = statali_2020.set_index('CODICESCUOLA')
statali_2021 = statali_2021.set_index('CODICESCUOLA')

In [9]:
# the process is repeated with private schools
paritarie_2019 = pd.read_csv("SCUANAGRAFEPAR20181920190831.csv") 
paritarie_2020 = pd.read_csv("SCUANAGRAFEPAR20192020200831.csv") 
paritarie_2021 = pd.read_csv("SCUANAGRAFEPAR20202120210831.csv") 

In [10]:
paritarie_2019 = paritarie_2019.set_index('CODICESCUOLA')
paritarie_2020 = paritarie_2020.set_index('CODICESCUOLA')
paritarie_2021 = paritarie_2021.set_index('CODICESCUOLA')

In [11]:
# three differents datasets are created, they can help to do a year-by-year analysis

scuole_2019 = pd.concat([statali_2019, paritarie_2019])
scuole_2020 = pd.concat([statali_2020, paritarie_2020])
scuole_2021 = pd.concat([statali_2021, paritarie_2021])

In [12]:
# we could eventually concatenate all datasets into one, but we should remember to eliminate duplicates in the index

#scuole = pd.concat([statali_2019, statali_2020, statali_2021, paritarie_2019, paritarie_2020, paritarie_2021])

In [13]:
scuole_2019.info()

<class 'pandas.core.frame.DataFrame'>
Index: 63942 entries, SAAA8AN034 to PD1A00900E
Data columns (total 19 columns):
 #   Column                                     Non-Null Count  Dtype 
---  ------                                     --------------  ----- 
 0   ANNOSCOLASTICO                             63942 non-null  int64 
 1   AREAGEOGRAFICA                             63942 non-null  object
 2   REGIONE                                    63942 non-null  object
 3   PROVINCIA                                  63942 non-null  object
 4   CODICEISTITUTORIFERIMENTO                  51194 non-null  object
 5   DENOMINAZIONEISTITUTORIFERIMENTO           51194 non-null  object
 6   DENOMINAZIONESCUOLA                        63942 non-null  object
 7   INDIRIZZOSCUOLA                            63942 non-null  object
 8   CAPSCUOLA                                  63942 non-null  object
 9   CODICECOMUNESCUOLA                         63942 non-null  object
 10  DESCRIZIONECOMUNE        

In [14]:
scuole_2020.info()

<class 'pandas.core.frame.DataFrame'>
Index: 63767 entries, FGAA841028 to RM1A69200P
Data columns (total 19 columns):
 #   Column                                     Non-Null Count  Dtype 
---  ------                                     --------------  ----- 
 0   ANNOSCOLASTICO                             63767 non-null  int64 
 1   AREAGEOGRAFICA                             63767 non-null  object
 2   REGIONE                                    63767 non-null  object
 3   PROVINCIA                                  63767 non-null  object
 4   CODICEISTITUTORIFERIMENTO                  51164 non-null  object
 5   DENOMINAZIONEISTITUTORIFERIMENTO           51164 non-null  object
 6   DENOMINAZIONESCUOLA                        63767 non-null  object
 7   INDIRIZZOSCUOLA                            63767 non-null  object
 8   CAPSCUOLA                                  63767 non-null  object
 9   CODICECOMUNESCUOLA                         63767 non-null  object
 10  DESCRIZIONECOMUNE        

In [15]:
scuole_2021.info()

<class 'pandas.core.frame.DataFrame'>
Index: 63494 entries, LIAA00202C to NA1A56300V
Data columns (total 19 columns):
 #   Column                                     Non-Null Count  Dtype 
---  ------                                     --------------  ----- 
 0   ANNOSCOLASTICO                             63494 non-null  int64 
 1   AREAGEOGRAFICA                             63494 non-null  object
 2   REGIONE                                    63494 non-null  object
 3   PROVINCIA                                  63494 non-null  object
 4   CODICEISTITUTORIFERIMENTO                  51174 non-null  object
 5   DENOMINAZIONEISTITUTORIFERIMENTO           51174 non-null  object
 6   DENOMINAZIONESCUOLA                        63494 non-null  object
 7   INDIRIZZOSCUOLA                            63494 non-null  object
 8   CAPSCUOLA                                  63494 non-null  object
 9   CODICECOMUNESCUOLA                         63494 non-null  object
 10  DESCRIZIONECOMUNE        

Possiamo notare come, rispetto ai dataset delle scuole con i numeri degli studenti che frequentano le ore di religione, ci siano una differenza di circa 10000 osservazioni (un sesto del dataset).

## Calcoliamo il numero di studenti totali


In [16]:
#every file is a dataset, by year, of all students for every classroom
numeri_statali_2019 = pd.read_csv('ALUCORSOETASTA20181920190831.csv')
numeri_statali_2020 = pd.read_csv('ALUCORSOETASTA20192020200831.csv')
numeri_statali_2021 = pd.read_csv('ALUCORSOETASTA20202120210831.csv')

In [17]:
#we need to aggregate the numbers by school, using the school unique identifier
numeri_statali_2019 = numeri_statali_2019.groupby('CODICESCUOLA').sum()
numeri_statali_2020 = numeri_statali_2020.groupby('CODICESCUOLA').sum()
numeri_statali_2021 = numeri_statali_2021.groupby('CODICESCUOLA').sum()

In [18]:
#we do the same for private schools
numeri_paritarie_2019 = pd.read_csv('ALUCORSOETAPAR20181920190831.csv')
numeri_paritarie_2020 = pd.read_csv('ALUCORSOETAPAR20192020200831.csv')
numeri_paritarie_2021 = pd.read_csv('ALUCORSOETAPAR20202120210831.csv')

In [19]:
numeri_paritarie_2019 = numeri_paritarie_2019.groupby('CODICESCUOLA').sum()
numeri_paritarie_2020 = numeri_paritarie_2020.groupby('CODICESCUOLA').sum()
numeri_paritarie_2021 = numeri_paritarie_2021.groupby('CODICESCUOLA').sum()

In [20]:
#we concatenate all datasets (state and private) for every year
numeri_2019 = pd.concat([numeri_paritarie_2019, numeri_statali_2019])
numeri_2020 = pd.concat([numeri_paritarie_2020, numeri_statali_2020])
numeri_2021 = pd.concat([numeri_paritarie_2021, numeri_statali_2021])

In [21]:
#dropping unnecessary columns
numeri_2019 = numeri_2019['ALUNNI']
numeri_2020 = numeri_2020['ALUNNI']
numeri_2021 = numeri_2021['ALUNNI']

In [22]:
numeri_2019

CODICESCUOLA
AG1E00500B     52
AG1E006007    125
AG1EVA500E     33
AGPC4A5009     20
AGPM00500N     34
             ... 
VVTF01101Q    408
VVTF04000P    212
VVTH01000A    353
VVTL01101X    111
VVTL011519     60
Name: ALUNNI, Length: 31898, dtype: int64

In [23]:
numeri_2020

CODICESCUOLA
AG1E00500B     29
AG1E006007    111
AG1EVA500E     53
AGPC4A5009     23
AGPM00500N     28
             ... 
VVTF01101Q    423
VVTF04000P    207
VVTH01000A    364
VVTL01101X     87
VVTL011519    113
Name: ALUNNI, Length: 31820, dtype: int64

In [24]:
numeri_2021

CODICESCUOLA
AG1E006007    117
AG1EVA500E     66
AGPC4A5009     19
AGPM00500N     15
AGPSCE5001     24
             ... 
VVTF040504     85
VVTH01000A    357
VVTL01101X     83
VVTL011519    119
VVTN00201L     79
Name: ALUNNI, Length: 31836, dtype: int64

Le differenze tra i dataset qui si fanno più sostanziali: aggregando i numeri di studenti per codice scuola otteniamo un numero di scuole inferiore rispetto alle scuole che comunicano le ore di religione. Non abbiamo quindi i dati sul numero totale di studenti di istituti che pur hanno comunicato le ore di religione.

Questo dataset sembra essere il più debole e meno completo dei tre.

## Joining dei datasets

Con questa tipologia di *join* diamo più importanza a mantenere l'integrità del dataset delle scuole che hanno comunicato quanti studenti frequentano le ore di religione.

In [25]:
data_2019 = data_2019.join(scuole_2019, how='left')
data_2019 = data_2019.join(numeri_2019, how='left')

In [26]:
data_2020 = data_2020.join(scuole_2020, how='left')
data_2020 = data_2020.join(numeri_2020, how='left')

In [27]:
data_2021 = data_2021.join(scuole_2021, how='left')
data_2021 = data_2021.join(numeri_2021, how='left')

## Analisi dati 2018/2019

Per analizzare il rapporto tra gli studenti totali e chi frequenta l'ora di religione dobbiamo necessariamente considerare solo le scuole che hanno un numero totale di studenti consolidato.

In [28]:
# drop columns which don't have the total number of students, as we can't use them for the analysis

data_2019 = data_2019[data_2019['ALUNNI'].notna()]

In [29]:
data_2019

Numero di alunni che frequentano l'ora di religione - a.s. 2018/19  \
Codice Scuola                                                                      
AG1E00500B                                                    52                   
AG1E006007                                                   125                   
AG1EVA500E                                                    32                   
AGEE01102E                                                   <=3                   
AGEE01104L                                                   359                   
...                                                          ...                   
VVTD01004R                                                    77                   
VVTF01101Q                                                   411                   
VVTF04000P                                                   170                   
VVTH01000A                                                   353                   
VVTL01101X                                                   112                   

               ANNOSCOLASTICO AREAGEOGRAFICA   REGIONE      PROVINCIA  \
Codice Scuola                                                           
AG1E00500B          201819.00          ISOLE   SICILIA      AGRIGENTO   
AG1E006007          201819.00          ISOLE   SICILIA      AGRIGENTO   
AG1EVA500E          201819.00          ISOLE   SICILIA      AGRIGENTO   
AGEE01102E          201819.00          ISOLE   SICILIA      AGRIGENTO   
AGEE01104L          201819.00          ISOLE   SICILIA      AGRIGENTO   
...                       ...            ...       ...            ...   
VVTD01004R          201819.00            SUD  CALABRIA  VIBO VALENTIA   
VVTF01101Q          201819.00            SUD  CALABRIA  VIBO VALENTIA   
VVTF04000P          201819.00            SUD  CALABRIA  VIBO VALENTIA   
VVTH01000A          201819.00            SUD  CALABRIA  VIBO VALENTIA   
VVTL01101X          201819.00            SUD  CALABRIA  VIBO VALENTIA   

              CODICEISTITUTORIFERIMENTO  \
Codice Scuola                             
AG1E00500B                          NaN   
AG1E006007                          NaN   
AG1EVA500E                          NaN   
AGEE01102E                   AGEE01100C   
AGEE01104L                   AGEE01100C   
...                                 ...   
VVTD01004R                   VVTD01000L   
VVTF01101Q                   VVIS011007   
VVTF04000P                   VVTF04000P   
VVTH01000A                   VVTH01000A   
VVTL01101X                   VVIS011007   

                    DENOMINAZIONEISTITUTORIFERIMENTO  \
Codice Scuola                                          
AG1E00500B                                       NaN   
AG1E006007                                       NaN   
AG1EVA500E                                       NaN   
AGEE01102E                            CD - DON BOSCO   
AGEE01104L                            CD - DON BOSCO   
...                                              ...   
VVTD01004R     ISTITUTO TECNICO ECONOMICO G. GALILEI   
VVTF01101Q                    I.I.S. I.T.G. E I.T.I.   
VVTF04000P                            ITIS "A.RUSSO"   
VVTH01000A                                       ITN   
VVTL01101X                    I.I.S. I.T.G. E I.T.I.   

                                        DENOMINAZIONESCUOLA  \
Codice Scuola                                                 
AG1E00500B       ISTITUTO ANCELLE RIPARATRICI "SACRO CUORE"   
AG1E006007     ENTE ECL.G.R.-SUORE VOCAZ."MARIA IMMACOLATA"   
AG1EVA500E                  ISTITUTO BILINGUE DON MORINELLO   
AGEE01102E                        PADRE GIOACCHINO LA LOMIA   
AGEE01104L                                        DON BOSCO   
...                                                     ...   
VVTD01004R                       ISTITUTO TECNICO ECONOMICO   
VVTF01101Q            IST TECNICO INDUSTRIALE VIBO VALENTIA   
VVTF04000P                                   ITIS "A

### Pulizia del dataset

Nel dataset sono riportati valori indicati con **<=3**, si è scelto di considerarli come **2** nell'analisi

In [30]:
data_2019["Numero di alunni che frequentano l'ora di religione - a.s. 2018/19"] = data_2019["Numero di alunni che frequentano l'ora di religione - a.s. 2018/19"].replace(['<=3'],'2')


In [31]:
#converted to floats
data_2019["Numero di alunni che frequentano l'ora di religione - a.s. 2018/19"] = pd.to_numeric(data_2019["Numero di alunni che frequentano l'ora di religione - a.s. 2018/19"])

In [32]:
data_2019['differenza'] = data_2019['ALUNNI'] - data_2019["Numero di alunni che frequentano l'ora di religione - a.s. 2018/19"]

In [33]:
data_2019['percentuale di chi frequenta ore di religione'] = data_2019["Numero di alunni che frequentano l'ora di religione - a.s. 2018/19"] / data_2019['ALUNNI'] * 100

In [34]:
#renaming columns
data_2019 = data_2019.rename(columns={"Numero di alunni che frequentano l'ora di religione - a.s. 2018/19": 'alunni che frequentano', "ALUNNI": 'totale alunni', 'differenza': "alunni che non frequentano"})

### Analisi numeriche

In [35]:
data_2019.groupby('AREAGEOGRAFICA').mean()

alunni che frequentano  ANNOSCOLASTICO  totale alunni  \
AREAGEOGRAFICA                                                          
CENTRO                          195.84       201819.00         234.42   
ISOLE                           189.71       201819.00         201.60   
NORD EST                        192.29       201819.00         237.22   
NORD OVEST                      181.12       201819.00         227.18   
SUD                             204.39       201819.00         213.97   

                alunni che non frequentano  \
AREAGEOGRAFICA                               
CENTRO                               38.59   
ISOLE                                11.89   
NORD EST                             44.93   
NORD OVEST                           46.07   
SUD                                   9.57   

                percentuale di chi frequenta ore di religione  
AREAGEOGRAFICA                                                 
CENTRO                                                  85.76  
ISOLE                                                   92.49  
NORD EST                                                83.83  
NORD OVEST                                              83.45  
SUD                                                     93.55

In [36]:
data_2019.groupby('REGIONE').mean().sort_values(by='percentuale di chi frequenta ore di religione', ascending = False)

alunni che frequentano  ANNOSCOLASTICO  totale alunni  \
REGIONE                                                                    
MOLISE                             127.56       201819.00         131.41   
BASILICATA                         145.00       201819.00         149.77   
CALABRIA                           139.92       201819.00         145.24   
PUGLIA                             283.86       201819.00         295.93   
SARDEGNA                           152.77       201819.00         168.63   
SICILIA                            202.89       201819.00         213.36   
ABRUZZO                            173.24       201819.00         186.44   
CAMPANIA                           215.71       201819.00         226.23   
UMBRIA                             173.55       201819.00         196.32   
LAZIO                              220.13       201819.00         254.28   
VENETO                             187.11       201819.00         221.99   
MARCHE                             176.71       201819.00         202.51   
LIGURIA                            154.12       201819.00         196.63   
PIEMONTE                           160.77       201819.00         203.66   
FRIULI-VENEZIA G.                  150.34       201819.00         188.17   
LOMBARDIA                          195.67       201819.00         243.90   
TOSCANA                            175.76       201819.00         231.50   
EMILIA ROMAGNA                     215.17       201819.00         276.70   

                   alunni che non frequentano  \
REGIONE                                         
MOLISE                                   3.85   
BASILICATA                               4.76   
CALABRIA                                 5.32   
PUGLIA                                  12.08   
SARDEGNA                                15.86   
SICILIA                                 10.47   
ABRUZZO                                 13.19   
CAMPANIA                                10.52   
UMBRIA                                  22.77   
LAZIO                                   34.15   
VENETO                                  34.88   
MARCHE                                  25.79   
LIGURIA                                 42.51   
PIEMONTE                                42.89   
FRIULI-VENEZIA G.                       37.83   
LOMBARDIA                               48.23   
TOSCANA                                 55.74   
EMILIA ROMAGNA                          61.53   

                   percentuale di chi frequenta ore di religione  
REGIONE                                                           
MOLISE                                                     97.50  
BASILICATA                                                 97.03  
CALABRIA                                                   95.56  
PUGLIA                                                     95.18  
SARDEGNA                                                   93.02  
SICILIA                                                    92.30  
ABRUZZO                                                    91.68  
CAMPANIA                                                   91.55  
UMBRIA                                                     89.03  
LAZIO                                                      87.83  
VENETO                                                     86.25  
MARCHE                                                     85.76  
LIGURIA                                                    84.44  
PIEMONTE                                                   83.82  
FRIULI-VENEZIA G.                                          83.46  
LOMBARDIA                                                  83.10  
TOSCANA                                                    81.80  
EMILIA ROMAGNA                                             80.62

In [37]:
data_2019.groupby('PROVINCIA').mean().sort_values(by='percentuale di chi frequenta ore di religione', ascending = False)[0:20]

alunni che frequentano  ANNOSCOLASTICO  totale alunni  \
PROVINCIA                                                                      
BARLETTA-ANDRIA-TRANI                  390.33       201819.00         395.52   
CAMPOBASSO                             136.84       201819.00         141.01   
ISERNIA                                105.55       201819.00         108.67   
POTENZA                                129.10       201819.00         132.91   
ORISTANO                               126.44       201819.00         135.96   
LECCE                                  222.81       201819.00         233.36   
VIBO VALENTIA                          115.98       201819.00         119.14   
CALTANISSETTA                          236.47       201819.00         243.24   
MATERA                                 184.08       201819.00         191.17   
CROTONE                                161.33       201819.00         167.94   
FOGGIA                                 235.82       201819.00         245.78   
CATANZARO                              136.40       201819.00         142.11   
MESSINA                                133.55       201819.00         138.32   
COSENZA                                127.34       201819.00         131.88   
ENNA                                   159.81       201819.00         165.06   
FROSINONE                              160.06       201819.00         165.96   
SUD SARDEGNA                           117.39       201819.00         123.09   
AVELLINO                               128.09       201819.00         134.45   
BRINDISI                               283.23       201819.00         294.34   
CASERTA                                221.49       201819.00         228.92   

                       alunni che non frequentano  \
PROVINCIA                                           
BARLETTA-ANDRIA-TRANI                        5.19   
CAMPOBASSO                                   4.16   
ISERNIA                                      3.12   
POTENZA                                      3.82   
ORISTANO                                     9.52   
LECCE                                       10.55   
VIBO VALENTIA                                3.16   
CALTANISSETTA                                6.76   
MATERA                                       7.09   
CROTONE                                      6.61   
FOGGIA                                       9.96   
CATANZARO                                    5.71   
MESSINA                                      4.77   
COSENZA                                      4.55   
ENNA                                         5.25   
FROSINONE                                    5.90   
SUD SARDEGNA                                 5.70   
AVELLINO                                     6.37   
BRINDISI                                    11.12   
CASERTA                                      7.43   

                       percentuale di chi frequenta ore di religione  
PROVINCIA                                                             
BARLETTA-ANDRIA-TRANI                                          98.39  
CAMPOBASSO                                                     97.57  
ISERNIA                                                        97.33  
POTENZA                                                        97.29  
ORISTANO                                                       96.97  
LECCE                                                          96.86  
VIBO VALENTIA                                                  96.80  
CALTANISSETTA                                                  96.72  
MATERA                                                         96.40  
CROTONE                                                        96.27  
FOGGIA                                                         96.04  
CATANZARO                                                      95.84  
MESSINA                                                        95.66  
COSENZA                       

In [38]:
data_2019.groupby('PROVINCIA').mean().sort_values(by='percentuale di chi frequenta ore di religione', ascending = True)[0:20]

alunni che frequentano  ANNOSCOLASTICO  totale alunni  \
PROVINCIA                                                              
BOLOGNA                        195.86       201819.00         284.69   
FIRENZE                        177.20       201819.00         272.23   
PRATO                          240.90       201819.00         336.41   
TRIESTE                        144.36       201819.00         207.34   
GORIZIA                        125.18       201819.00         168.48   
MODENA                         242.87       201819.00         319.16   
LIVORNO                        212.29       201819.00         279.50   
GROSSETO                       123.37       201819.00         154.62   
FERMO                          139.72       201819.00         165.38   
REGGIO EMILIA                  202.29       201819.00         260.67   
TORINO                         188.07       201819.00         248.24   
MANTOVA                        167.93       201819.00         217.05   
MILANO                         242.15       201819.00         314.01   
PIACENZA                       172.86       201819.00         219.75   
BRESCIA                        176.10       201819.00         226.69   
FERRARA                        185.98       201819.00         238.95   
RAVENNA                        247.11       201819.00         316.88   
BERGAMO                        175.82       201819.00         218.87   
BIELLA                         111.43       201819.00         151.62   
SIENA                          165.64       201819.00         213.85   

               alunni che non frequentano  \
PROVINCIA                                   
BOLOGNA                             88.83   
FIRENZE                             95.03   
PRATO                               95.51   
TRIESTE                             62.98   
GORIZIA                             43.30   
MODENA                              76.28   
LIVORNO                             67.21   
GROSSETO                            31.25   
FERMO                               25.66   
REGGIO EMILIA                       58.38   
TORINO                              60.17   
MANTOVA                             49.12   
MILANO                              71.86   
PIACENZA                            46.89   
BRESCIA                             50.58   
FERRARA                             52.97   
RAVENNA                             69.76   
BERGAMO                             43.06   
BIELLA                              40.18   
SIENA                               48.21   

               percentuale di chi frequenta ore di religione  
PROVINCIA                                                     
BOLOGNA                                                73.91  
FIRENZE                                                73.94  
PRATO                                                  75.03  
TRIESTE                                                75.13  
GORIZIA                                                76.14  
MODENA                                                 78.00  
LIVORNO                                                78.05  
GROSSETO                                               79.67  
FERMO                                                  79.86  
REGGIO EMILIA                                          79.94  
TORINO                                                 80.06  
MANTOVA                                                80.10  
MILANO                                                 80.34  
PIACENZA                                               80.74  
BRESCIA                                                81.17  
FERRARA                                                81.41  
RAVENNA                                                82.11  
BERGAMO                                                82.11  
BIELLA                                                 82.14  
SIENA                                                  82.44

In [39]:
data_2019.groupby('DESCRIZIONETIPOLOGIAGRADOISTRUZIONESCUOLA').mean().sort_values(by='percentuale di chi frequenta ore di religione', ascending = False)

alunni che frequentano  \
DESCRIZIONETIPOLOGIAGRADOISTRUZIONESCUOLA                                   
IST PROF INDUSTRIA E ARTIGIANATO PER CIECHI                        228.00   
SCUOLA PRIMARIA NON STATALE                                        114.91   
SCUOLA SEC. PRIMO GRADO NON STATALE                                 99.76   
SCUOLA PRIMARIA                                                    151.01   
SCUOLA PRIMO GRADO                                                 197.64   
LICEO CLASSICO                                                     490.23   
ISTITUTO TECNICO NAUTICO                                           382.68   
IST PROF PER L'AGRICOLTURA E L'AMBIENTE                            162.03   
ISTITUTO TECNICO AGRARIO                                           260.83   
LICEO SCIENTIFICO                                                  538.67   
ISTITUTO MAGISTRALE                                                573.97   
IST TEC COMMERCIALE E PER GEOMETRI                                 347.20   
IST PROF PER I SERVIZI ALBERGHIERI E RISTORAZIONE                  476.18   
ISTITUTO TECNICO PER GEOMETRI                                      177.95   
IST PROF PER I SERVIZI SOCIALI                                     266.47   
IST PROF PER L'AGRICOLTURA                                         150.88   
IST PROF INDUSTRIA E ATTIVITA' MARINARE                            120.33   
ISTITUTO TECNICO COMMERCIALE                                       303.03   
ISTITUTO D'ARTE                                                    282.49   
SCUOLA MAGISTRALE                                                  315.25   
IST PROF ALBERGHIERO                                               227.75   
ISTITUTO TECNICO INDUSTRIALE                                       445.62   
LICEO ARTISTICO                                                    304.79   
ISTITUTO TECNICO PER ATTIVITA' SOCIALI (GIA' ITF)                  374.17   
ISTITUTO TECNICO PER IL TURISMO                                    268.02   
IST PROF INDUSTRIA E ARTIGIANATO                                   195.23   
IST PROF PER I SERVIZI COMMERCIALI                                 213.26   
IST PROF INDUSTRIA E ARTIGIANATO PER SORDOMUTI                     106.50   
IST PROF PER I SERVIZI COMMERCIALI E TURISTICI                     250.32   
ISTITUTO TECNICO AERONAUTICO                                       469.00   
IST PROF PER I SERVIZI COMM TUR E DELLA PUBB                       116.00   
IST PROF PER I SERVIZI TURISTICI                                   330.00   
IST PROF PER I SERVIZI PUBBLICITARI                                295.00   
IST PROF CINEMATOGRAFIA E TELEVISIONE                              524.00   
SCUOLA SEC. SECONDO GRADO NON STATALE                               42.68   

                                                   ANNOSCOLASTICO  \
DESCRIZIONETIPOLOGIAGRADOISTRUZIONESCUOLA                           
IST PROF INDUSTRIA E ARTIGIANATO PER CIECHI             201819.00   
SCUOLA PRIMARIA NON STATALE                             201819.00   
SCUOLA SEC. PRIMO GRADO NON STATALE                     201819.00   
SCUOLA PRIMARIA                                         201819.00   
SCUOLA PRIMO GRADO                                      201819.00   
LICEO CLASSICO                                          201819.00   
ISTITUTO TECNICO NAUTICO                                201819.00   
IST PROF PER L'AGRICOLTURA E L'AMBIENTE                 201819.00   
ISTITUTO TECNICO AGRARIO                                201819.00   
LICEO SCIENTIFICO                                       201819.00   
ISTITUTO MAGISTRALE                                     201819.00   
IST TEC COMMERCIALE E PER GEOMETRI                      201819.00   
IST PROF PER I SERVIZI ALBERGHIERI E RISTORAZIONE       201819.00   
ISTITUTO TECNICO PER GEOMETRI                           201819.00   
IST PROF PER I SERVIZI SOCIALI                          201819.00   
IST PROF PER L'AGRICOLTURA  

## Analisi dati 2019/2020

Valgono le stesse considerazioni fatte per l'analisi 2018/2019.

In [40]:
data_2020 = data_2020[data_2020['ALUNNI'].notna()]

In [41]:
data_2020

Numero di alunni che frequentano l'ora di religione - a.s. 2019/20  \
Codice Scuola                                                                      
AG1E00500B                                                    31                   
AG1E006007                                                   112                   
AG1EVA500E                                                    51                   
AGEE01102E                                                   242                   
AGEE01104L                                                   352                   
...                                                          ...                   
VVTD01004R                                                    84                   
VVTF01101Q                                                   423                   
VVTF04000P                                                   204                   
VVTH01000A                                                   362                   
VVTL01101X                                                    86                   

               ANNOSCOLASTICO AREAGEOGRAFICA   REGIONE      PROVINCIA  \
Codice Scuola                                                           
AG1E00500B          201920.00          ISOLE   SICILIA      AGRIGENTO   
AG1E006007          201920.00          ISOLE   SICILIA      AGRIGENTO   
AG1EVA500E          201920.00          ISOLE   SICILIA      AGRIGENTO   
AGEE01102E          201920.00          ISOLE   SICILIA      AGRIGENTO   
AGEE01104L          201920.00          ISOLE   SICILIA      AGRIGENTO   
...                       ...            ...       ...            ...   
VVTD01004R          201920.00            SUD  CALABRIA  VIBO VALENTIA   
VVTF01101Q          201920.00            SUD  CALABRIA  VIBO VALENTIA   
VVTF04000P          201920.00            SUD  CALABRIA  VIBO VALENTIA   
VVTH01000A          201920.00            SUD  CALABRIA  VIBO VALENTIA   
VVTL01101X          201920.00            SUD  CALABRIA  VIBO VALENTIA   

              CODICEISTITUTORIFERIMENTO  \
Codice Scuola                             
AG1E00500B                          NaN   
AG1E006007                          NaN   
AG1EVA500E                          NaN   
AGEE01102E                   AGEE01100C   
AGEE01104L                   AGEE01100C   
...                                 ...   
VVTD01004R                   VVTD01000L   
VVTF01101Q                   VVIS011007   
VVTF04000P                   VVTF04000P   
VVTH01000A                   VVTH01000A   
VVTL01101X                   VVIS011007   

                    DENOMINAZIONEISTITUTORIFERIMENTO  \
Codice Scuola                                          
AG1E00500B                                       NaN   
AG1E006007                                       NaN   
AG1EVA500E                                       NaN   
AGEE01102E                            CD - DON BOSCO   
AGEE01104L                            CD - DON BOSCO   
...                                              ...   
VVTD01004R     ISTITUTO TECNICO ECONOMICO G. GALILEI   
VVTF01101Q                    I.I.S. I.T.G. E I.T.I.   
VVTF04000P                            ITIS "A.RUSSO"   
VVTH01000A                                       ITN   
VVTL01101X                    I.I.S. I.T.G. E I.T.I.   

                                        DENOMINAZIONESCUOLA  \
Codice Scuola                                                 
AG1E00500B       ISTITUTO ANCELLE RIPARATRICI "SACRO CUORE"   
AG1E006007     ENTE ECL.G.R.-SUORE VOCAZ."MARIA IMMACOLATA"   
AG1EVA500E                  ISTITUTO BILINGUE DON MORINELLO   
AGEE01102E                        PADRE GIOACCHINO LA LOMIA   
AGEE01104L                                        DON BOSCO   
...                                                     ...   
VVTD01004R                       ISTITUTO TECNICO ECONOMICO   
VVTF01101Q            IST TECNICO INDUSTRIALE VIBO VALENTIA   
VVTF04000P                                   ITIS "A

### Pulizia del dataset

Nel dataset sono riportati valori indicati con **<=3**, si è scelto di considerarli come **2** nell'analisi.

In [43]:
data_2020["Numero di alunni che frequentano l'ora di religione - a.s. 2019/20"] = data_2020["Numero di alunni che frequentano l'ora di religione - a.s. 2019/20"].replace(['<=3'],'2')


In [45]:
#converted to floats
data_2020["Numero di alunni che frequentano l'ora di religione - a.s. 2019/20"] = pd.to_numeric(data_2020["Numero di alunni che frequentano l'ora di religione - a.s. 2019/20"])

In [47]:
data_2020['differenza'] = data_2020['ALUNNI'] - data_2020["Numero di alunni che frequentano l'ora di religione - a.s. 2019/20"]

In [48]:
data_2020['percentuale di chi frequenta ore di religione'] = data_2020["Numero di alunni che frequentano l'ora di religione - a.s. 2019/20"] / data_2020['ALUNNI'] * 100

In [49]:
#renaming columns
data_2020 = data_2020.rename(columns={"Numero di alunni che frequentano l'ora di religione - a.s. 2019/20": 'alunni che frequentano', "ALUNNI": 'totale alunni', 'differenza': "alunni che non frequentano"})

### Analisi numeriche

In [50]:
data_2020.groupby('AREAGEOGRAFICA').mean()

alunni che frequentano  ANNOSCOLASTICO  totale alunni  \
AREAGEOGRAFICA                                                          
CENTRO                          193.13       201920.00         234.01   
ISOLE                           188.45       201920.00         199.70   
NORD EST                        189.57       201920.00         237.00   
NORD OVEST                      178.56       201920.00         227.03   
SUD                             201.63       201920.00         212.14   

                alunni che non frequentano  \
AREAGEOGRAFICA                               
CENTRO                               40.89   
ISOLE                                11.24   
NORD EST                             47.43   
NORD OVEST                           48.47   
SUD                                  10.51   

                percentuale di chi frequenta ore di religione  
AREAGEOGRAFICA                                                 
CENTRO                                                  84.97  
ISOLE                                                   92.77  
NORD EST                                                83.04  
NORD OVEST                                              82.47  
SUD                                                     93.16

In [51]:
data_2020.groupby('REGIONE').mean().sort_values(by='percentuale di chi frequenta ore di religione', ascending = False)

alunni che frequentano  ANNOSCOLASTICO  totale alunni  \
REGIONE                                                                    
BASILICATA                         141.99       201920.00         147.50   
MOLISE                             127.19       201920.00         131.39   
PUGLIA                             289.10       201920.00         298.00   
CALABRIA                           137.62       201920.00         143.42   
SARDEGNA                           151.69       201920.00         167.04   
SICILIA                            201.60       201920.00         211.37   
ABRUZZO                            171.99       201920.00         185.08   
CAMPANIA                           209.13       201920.00         223.00   
UMBRIA                             171.62       201920.00         195.47   
LAZIO                              216.60       201920.00         253.44   
MARCHE                             176.66       201920.00         201.96   
VENETO                             184.20       201920.00         221.42   
LIGURIA                            150.97       201920.00         196.79   
FRIULI-VENEZIA G.                  148.80       201920.00         187.86   
PIEMONTE                           157.93       201920.00         202.47   
LOMBARDIA                          193.37       201920.00         244.22   
TOSCANA                            172.63       201920.00         231.85   
EMILIA ROMAGNA                     212.16       201920.00         276.79   

                   alunni che non frequentano  \
REGIONE                                         
BASILICATA                               5.52   
MOLISE                                   4.20   
PUGLIA                                   8.91   
CALABRIA                                 5.80   
SARDEGNA                                15.35   
SICILIA                                  9.77   
ABRUZZO                                 13.09   
CAMPANIA                                13.87   
UMBRIA                                  23.85   
LAZIO                                   36.84   
MARCHE                                  25.30   
VENETO                                  37.22   
LIGURIA                                 45.82   
FRIULI-VENEZIA G.                       39.06   
PIEMONTE                                44.53   
LOMBARDIA                               50.85   
TOSCANA                                 59.23   
EMILIA ROMAGNA                          64.63   

                   percentuale di chi frequenta ore di religione  
REGIONE                                                           
BASILICATA                                                 97.26  
MOLISE                                                     97.14  
PUGLIA                                                     96.15  
CALABRIA                                                   95.16  
SARDEGNA                                                   93.18  
SICILIA                                                    92.62  
ABRUZZO                                                    91.51  
CAMPANIA                                                   90.45  
UMBRIA                                                     88.55  
LAZIO                                                      86.73  
MARCHE                                                     85.67  
VENETO                                                     85.49  
LIGURIA                                                    83.02  
FRIULI-VENEZIA G.                                          82.93  
PIEMONTE                                                   82.88  
LOMBARDIA                                                  82.17  
TOSCANA                                                    81.02  
EMILIA ROMAGNA                                             79.70

In [52]:
data_2020.groupby('PROVINCIA').mean().sort_values(by='percentuale di chi frequenta ore di religione', ascending = False)[0:20]

alunni che frequentano  ANNOSCOLASTICO  totale alunni  \
PROVINCIA                                                                      
POTENZA                                126.26       201920.00         130.37   
ISERNIA                                109.34       201920.00         112.29   
BRINDISI                               305.08       201920.00         311.33   
ORISTANO                               125.51       201920.00         133.61   
BARLETTA-ANDRIA-TRANI                  392.89       201920.00         399.36   
VIBO VALENTIA                          112.15       201920.00         115.68   
LECCE                                  224.93       201920.00         231.39   
CAMPOBASSO                             134.54       201920.00         139.26   
FOGGIA                                 248.36       201920.00         256.45   
ENNA                                   161.65       201920.00         164.85   
MESSINA                                132.44       201920.00         137.32   
MATERA                                 180.86       201920.00         189.86   
CALTANISSETTA                          232.20       201920.00         241.50   
COSENZA                                126.13       201920.00         131.02   
CROTONE                                155.68       201920.00         163.35   
CATANZARO                              136.08       201920.00         142.27   
SUD SARDEGNA                           115.06       201920.00         121.49   
FROSINONE                              157.82       201920.00         164.40   
BARI                                   344.13       201920.00         358.04   
NUORO                                  114.19       201920.00         122.58   

                       alunni che non frequentano  \
PROVINCIA                                           
POTENZA                                      4.11   
ISERNIA                                      2.94   
BRINDISI                                     6.25   
ORISTANO                                     8.11   
BARLETTA-ANDRIA-TRANI                        6.47   
VIBO VALENTIA                                3.54   
LECCE                                        6.46   
CAMPOBASSO                                   4.72   
FOGGIA                                       8.08   
ENNA                                         3.19   
MESSINA                                      4.88   
MATERA                                       9.00   
CALTANISSETTA                                9.31   
COSENZA                                      4.90   
CROTONE                                      7.67   
CATANZARO                                    6.19   
SUD SARDEGNA                                 6.44   
FROSINONE                                    6.58   
BARI                                        13.91   
NUORO                                        8.39   

                       percentuale di chi frequenta ore di religione  
PROVINCIA                                                             
POTENZA                                                        97.90  
ISERNIA                                                        97.70  
BRINDISI                                                       97.61  
ORISTANO                                                       97.42  
BARLETTA-ANDRIA-TRANI                                          97.42  
VIBO VALENTIA                                                  97.42  
LECCE                                                          97.15  
CAMPOBASSO                                                     96.91  
FOGGIA                                                         96.60  
ENNA                                                           96.37  
MESSINA                                                        95.71  
MATERA                                                         95.67  
CALTANISSETTA                                                  95.54  
COSENZA                       

In [53]:
data_2020.groupby('PROVINCIA').mean().sort_values(by='percentuale di chi frequenta ore di religione', ascending = True)[0:20]

alunni che frequentano  ANNOSCOLASTICO  totale alunni  \
PROVINCIA                                                              
FIRENZE                        171.05       201920.00         272.24   
BOLOGNA                        194.86       201920.00         285.43   
PRATO                          237.38       201920.00         341.46   
TRIESTE                        143.21       201920.00         206.93   
GORIZIA                        125.12       201920.00         171.29   
MODENA                         237.24       201920.00         316.68   
LIVORNO                        209.26       201920.00         281.89   
MANTOVA                        166.02       201920.00         219.10   
FERMO                          138.55       201920.00         161.96   
TORINO                         184.34       201920.00         246.14   
FERRARA                        184.14       201920.00         240.15   
MILANO                         239.33       201920.00         314.55   
IMPERIA                        123.30       201920.00         158.27   
REGGIO EMILIA                  200.06       201920.00         260.64   
GROSSETO                       120.46       201920.00         153.55   
BRESCIA                        172.17       201920.00         226.26   
PIACENZA                       170.54       201920.00         220.70   
BIELLA                         106.52       201920.00         147.76   
BERGAMO                        170.83       201920.00         216.95   
RAVENNA                        241.66       201920.00         315.90   

               alunni che non frequentano  \
PROVINCIA                                   
FIRENZE                            101.19   
BOLOGNA                             90.57   
PRATO                              104.07   
TRIESTE                             63.72   
GORIZIA                             46.17   
MODENA                              79.44   
LIVORNO                             72.64   
MANTOVA                             53.08   
FERMO                               23.41   
TORINO                              61.79   
FERRARA                             56.01   
MILANO                              75.22   
IMPERIA                             34.97   
REGGIO EMILIA                       60.58   
GROSSETO                            33.09   
BRESCIA                             54.10   
PIACENZA                            50.17   
BIELLA                              41.23   
BERGAMO                             46.12   
RAVENNA                             74.24   

               percentuale di chi frequenta ore di religione  
PROVINCIA                                                     
FIRENZE                                                73.14  
BOLOGNA                                                73.30  
PRATO                                                  73.90  
TRIESTE                                                74.90  
GORIZIA                                                75.18  
MODENA                                                 76.78  
LIVORNO                                                76.85  
MANTOVA                                                78.28  
FERMO                                                  78.38  
TORINO                                                 79.27  
FERRARA                                                79.38  
MILANO                                                 79.46  
IMPERIA                                                79.56  
REGGIO EMILIA                                          79.57  
GROSSETO                                               79.67  
BRESCIA                                                79.71  
PIACENZA                                               80.19  
BIELLA                                                 81.10  
BERGAMO                                                81.14  
RAVENNA                                                81.72

In [54]:
data_2020.groupby('DESCRIZIONETIPOLOGIAGRADOISTRUZIONESCUOLA').mean().sort_values(by='percentuale di chi frequenta ore di religione', ascending = False)

alunni che frequentano  \
DESCRIZIONETIPOLOGIAGRADOISTRUZIONESCUOLA                                   
IST PROF INDUSTRIA E ARTIGIANATO PER CIECHI                        208.00   
SCUOLA PRIMARIA NON STATALE                                        114.26   
SCUOLA PRIMARIA                                                    147.66   
SCUOLA SEC. PRIMO GRADO NON STATALE                                100.60   
SCUOLA PRIMO GRADO                                                 195.90   
LICEO CLASSICO                                                     500.57   
ISTITUTO TECNICO NAUTICO                                           375.40   
IST PROF PER L'AGRICOLTURA E L'AMBIENTE                            154.04   
ISTITUTO TECNICO AGRARIO                                           241.74   
LICEO SCIENTIFICO                                                  538.88   
IST PROF PER L'AGRICOLTURA                                         144.00   
ISTITUTO MAGISTRALE                                                574.08   
SCUOLA MAGISTRALE                                                  333.50   
IST TEC COMMERCIALE E PER GEOMETRI                                 345.96   
IST PROF PER I SERVIZI ALBERGHIERI E RISTORAZIONE                  429.21   
IST PROF PER I SERVIZI SOCIALI                                     268.82   
ISTITUTO D'ARTE                                                    281.34   
ISTITUTO TECNICO COMMERCIALE                                       294.32   
ISTITUTO TECNICO PER ATTIVITA' SOCIALI (GIA' ITF)                  382.73   
ISTITUTO TECNICO PER GEOMETRI                                      172.51   
IST PROF INDUSTRIA E ATTIVITA' MARINARE                            114.00   
ISTITUTO TECNICO INDUSTRIALE                                       445.52   
LICEO ARTISTICO                                                    305.90   
IST PROF ALBERGHIERO                                               213.00   
IST PROF INDUSTRIA E ARTIGIANATO                                   191.01   
IST PROF INDUSTRIA E ARTIGIANATO PER SORDOMUTI                     102.00   
ISTITUTO TECNICO PER IL TURISMO                                    260.42   
IST PROF PER I SERVIZI COMMERCIALI                                 211.99   
IST PROF PER I SERVIZI COMMERCIALI E TURISTICI                     248.89   
ISTITUTO TECNICO AERONAUTICO                                       487.67   
IST PROF PER I SERVIZI PUBBLICITARI                                313.50   
IST PROF PER I SERVIZI COMM TUR E DELLA PUBB                       112.00   
IST PROF PER I SERVIZI TURISTICI                                   309.75   
IST PROF CINEMATOGRAFIA E TELEVISIONE                              522.00   
SCUOLA SEC. SECONDO GRADO NON STATALE                               43.01   

                                                   ANNOSCOLASTICO  \
DESCRIZIONETIPOLOGIAGRADOISTRUZIONESCUOLA                           
IST PROF INDUSTRIA E ARTIGIANATO PER CIECHI             201920.00   
SCUOLA PRIMARIA NON STATALE                             201920.00   
SCUOLA PRIMARIA                                         201920.00   
SCUOLA SEC. PRIMO GRADO NON STATALE                     201920.00   
SCUOLA PRIMO GRADO                                      201920.00   
LICEO CLASSICO                                          201920.00   
ISTITUTO TECNICO NAUTICO                                201920.00   
IST PROF PER L'AGRICOLTURA E L'AMBIENTE                 201920.00   
ISTITUTO TECNICO AGRARIO                                201920.00   
LICEO SCIENTIFICO                                       201920.00   
IST PROF PER L'AGRICOLTURA                              201920.00   
ISTITUTO MAGISTRALE                                     201920.00   
SCUOLA MAGISTRALE                                       201920.00   
IST TEC COMMERCIALE E PER GEOMETRI                      201920.00   
IST PROF PER I SERVIZI ALBERGHIERI E RISTORAZIONE       201920.00   
IST PROF PER I SERVIZI SOCIA

## Analisi dati 2020/2021

Valgono le stesse considerazioni fatte per l'analisi 2018/2019.

In [55]:
data_2021 = data_2021[data_2021['ALUNNI'].notna()]

In [56]:
data_2021

Numero di alunni che frequentano l'ora di religione - a.s. 2020/21  \
Codice Scuola                                                                      
AG1E006007                                                   117                   
AG1EVA500E                                                    66                   
AGEE01102E                                                   250                   
AGEE01104L                                                   336                   
AGEE034016                                                   165                   
...                                                          ...                   
VVTF01101Q                                                   445                   
VVTF04000P                                                   212                   
VVTH01000A                                                   332                   
VVTL01101X                                                    84                   
VVTN00201L                                                    80                   

               ANNOSCOLASTICO AREAGEOGRAFICA   REGIONE      PROVINCIA  \
Codice Scuola                                                           
AG1E006007          202021.00          ISOLE   SICILIA      AGRIGENTO   
AG1EVA500E          202021.00          ISOLE   SICILIA      AGRIGENTO   
AGEE01102E          202021.00          ISOLE   SICILIA      AGRIGENTO   
AGEE01104L          202021.00          ISOLE   SICILIA      AGRIGENTO   
AGEE034016          202021.00          ISOLE   SICILIA      AGRIGENTO   
...                       ...            ...       ...            ...   
VVTF01101Q          202021.00            SUD  CALABRIA  VIBO VALENTIA   
VVTF04000P          202021.00            SUD  CALABRIA  VIBO VALENTIA   
VVTH01000A          202021.00            SUD  CALABRIA  VIBO VALENTIA   
VVTL01101X          202021.00            SUD  CALABRIA  VIBO VALENTIA   
VVTN00201L          202021.00            SUD  CALABRIA  VIBO VALENTIA   

              CODICEISTITUTORIFERIMENTO  \
Codice Scuola                             
AG1E006007                          NaN   
AG1EVA500E                          NaN   
AGEE01102E                   AGEE01100C   
AGEE01104L                   AGEE01100C   
AGEE034016                   AGEE034005   
...                                 ...   
VVTF01101Q                   VVIS011007   
VVTF04000P                   VVTF04000P   
VVTH01000A                   VVTH01000A   
VVTL01101X                   VVIS011007   
VVTN00201L                   VVIS00200C   

                    DENOMINAZIONEISTITUTORIFERIMENTO  \
Codice Scuola                                          
AG1E006007                                       NaN   
AG1EVA500E                                       NaN   
AGEE01102E                            CD - DON BOSCO   
AGEE01104L                            CD - DON BOSCO   
AGEE034016             DD I CIRCOLO - GIOVANNI XXIII   
...                                              ...   
VVTF01101Q                    I.I.S. I.T.G. E I.T.I.   
VVTF04000P                            ITIS "A.RUSSO"   
VVTH01000A                                       ITN   
VVTL01101X                    I.I.S. I.T.G. E I.T.I.   
VVTN00201L     IST D'ISTRUZIONE SUPERIORE P.GALLUPPI   

                                        DENOMINAZIONESCUOLA  \
Codice Scuola                                                 
AG1E006007     ENTE ECL.G.R.-SUORE VOCAZ."MARIA IMMACOLATA"   
AG1EVA500E                  ISTITUTO BILINGUE DON MORINELLO   
AGEE01102E                        PADRE GIOACCHINO LA LOMIA   
AGEE01104L                                        DON BOSCO   
AGEE034016                                          FAZELLO   
...                                                     ...   
VVTF01101Q            IST TECNICO INDUSTRIALE VIBO VALENTIA   
VVTF04000P                                   ITIS "A.RUSSO"   
VVTH01000A                                          

### Pulizia del dataset

Nel dataset sono riportati valori indicati con **<=3**, si è scelto di considerarli come **2** nell'analisi.

In [57]:
data_2021["Numero di alunni che frequentano l'ora di religione - a.s. 2020/21"] = data_2021["Numero di alunni che frequentano l'ora di religione - a.s. 2020/21"].replace(['<=3'],'2')


In [58]:
#converted to floats
data_2021["Numero di alunni che frequentano l'ora di religione - a.s. 2020/21"] = pd.to_numeric(data_2021["Numero di alunni che frequentano l'ora di religione - a.s. 2020/21"])

In [59]:
data_2021['differenza'] = data_2021['ALUNNI'] - data_2021["Numero di alunni che frequentano l'ora di religione - a.s. 2020/21"]

In [60]:
data_2021['percentuale di chi frequenta ore di religione'] = data_2021["Numero di alunni che frequentano l'ora di religione - a.s. 2020/21"] / data_2021['ALUNNI'] * 100

In [61]:
#renaming columns
data_2021 = data_2021.rename(columns={"Numero di alunni che frequentano l'ora di religione - a.s. 2020/21": 'alunni che frequentano', "ALUNNI": 'totale alunni', 'differenza': "alunni che non frequentano"})

### Analisi numeriche

In [62]:
data_2021.groupby('AREAGEOGRAFICA').mean()

alunni che frequentano  ANNOSCOLASTICO  totale alunni  \
AREAGEOGRAFICA                                                          
CENTRO                          191.65       202021.00         233.21   
ISOLE                           185.61       202021.00         197.89   
NORD EST                        187.38       202021.00         236.06   
NORD OVEST                      176.03       202021.00         225.89   
SUD                             198.18       202021.00         209.53   

                alunni che non frequentano  \
AREAGEOGRAFICA                               
CENTRO                               41.55   
ISOLE                                12.29   
NORD EST                             48.68   
NORD OVEST                           49.86   
SUD                                  11.35   

                percentuale di chi frequenta ore di religione  
AREAGEOGRAFICA                                                 
CENTRO                                                  84.59  
ISOLE                                                   92.29  
NORD EST                                                82.73  
NORD OVEST                                              82.04  
SUD                                                     92.58

In [63]:
data_2021.groupby('REGIONE').mean().sort_values(by='percentuale di chi frequenta ore di religione', ascending = False)

alunni che frequentano  ANNOSCOLASTICO  totale alunni  \
REGIONE                                                                    
MOLISE                             127.17       202021.00         131.15   
BASILICATA                         140.77       202021.00         145.31   
PUGLIA                             281.22       202021.00         293.36   
CALABRIA                           134.60       202021.00         142.34   
SARDEGNA                           148.70       202021.00         165.27   
SICILIA                            198.83       202021.00         209.58   
ABRUZZO                            170.23       202021.00         183.99   
CAMPANIA                           206.33       202021.00         219.74   
UMBRIA                             168.88       202021.00         194.46   
LAZIO                              216.52       202021.00         252.97   
MARCHE                             175.58       202021.00         201.58   
VENETO                             182.61       202021.00         220.07   
PIEMONTE                           155.15       202021.00         201.39   
FRIULI-VENEZIA G.                  145.68       202021.00         186.72   
LIGURIA                            147.40       202021.00         195.44   
LOMBARDIA                          191.15       202021.00         243.11   
TOSCANA                            169.25       202021.00         230.34   
EMILIA ROMAGNA                     209.37       202021.00         276.27   

                   alunni che non frequentano  \
REGIONE                                         
MOLISE                                   3.98   
BASILICATA                               4.53   
PUGLIA                                  12.14   
CALABRIA                                 7.73   
SARDEGNA                                16.58   
SICILIA                                 10.75   
ABRUZZO                                 13.77   
CAMPANIA                                13.41   
UMBRIA                                  25.58   
LAZIO                                   36.44   
MARCHE                                  26.00   
VENETO                                  37.46   
PIEMONTE                                46.23   
FRIULI-VENEZIA G.                       41.04   
LIGURIA                                 48.03   
LOMBARDIA                               51.96   
TOSCANA                                 61.09   
EMILIA ROMAGNA                          66.90   

                   percentuale di chi frequenta ore di religione  
REGIONE                                                           
MOLISE                                                     96.89  
BASILICATA                                                 96.53  
PUGLIA                                                     95.14  
CALABRIA                                                   94.25  
SARDEGNA                                                   92.67  
SICILIA                                                    92.16  
ABRUZZO                                                    90.56  
CAMPANIA                                                   90.32  
UMBRIA                                                     87.58  
LAZIO                                                      86.43  
MARCHE                                                     85.68  
VENETO                                                     85.47  
PIEMONTE                                                   82.55  
FRIULI-VENEZIA G.                                          82.28  
LIGURIA                                                    82.27  
LOMBARDIA                                                  81.75  
TOSCANA                                                    80.50  
EMILIA ROMAGNA                                             79.14

In [64]:
data_2021.groupby('PROVINCIA').mean().sort_values(by='percentuale di chi frequenta ore di religione', ascending = False)[0:20]

alunni che frequentano  ANNOSCOLASTICO  totale alunni  \
PROVINCIA                                                                      
VIBO VALENTIA                          107.11       202021.00         112.93   
BARLETTA-ANDRIA-TRANI                  377.67       202021.00         387.40   
CROTONE                                157.98       202021.00         163.95   
CAMPOBASSO                             133.90       202021.00         138.11   
POTENZA                                124.34       202021.00         127.65   
FOGGIA                                 243.17       202021.00         251.42   
ORISTANO                               122.58       202021.00         131.77   
BRINDISI                               298.74       202021.00         308.12   
CATANZARO                              136.33       202021.00         141.87   
ISERNIA                                110.53       202021.00         113.96   
ENNA                                   154.04       202021.00         161.18   
TARANTO                                287.00       202021.00         295.30   
CALTANISSETTA                          230.70       202021.00         239.21   
TRAPANI                                210.59       202021.00         219.05   
MATERA                                 181.80       202021.00         189.38   
NUORO                                  111.00       202021.00         120.83   
BENEVENTO                              125.35       202021.00         130.77   
RIETI                                  123.54       202021.00         134.03   
LECCE                                  213.18       202021.00         226.47   
SUD SARDEGNA                           111.90       202021.00         118.55   

                       alunni che non frequentano  \
PROVINCIA                                           
VIBO VALENTIA                                5.82   
BARLETTA-ANDRIA-TRANI                        9.73   
CROTONE                                      5.97   
CAMPOBASSO                                   4.20   
POTENZA                                      3.31   
FOGGIA                                       8.25   
ORISTANO                                     9.19   
BRINDISI                                     9.38   
CATANZARO                                    5.54   
ISERNIA                                      3.43   
ENNA                                         7.13   
TARANTO                                      8.29   
CALTANISSETTA                                8.51   
TRAPANI                                      8.45   
MATERA                                       7.59   
NUORO                                        9.83   
BENEVENTO                                    5.41   
RIETI                                       10.48   
LECCE                                       13.29   
SUD SARDEGNA                                 6.65   

                       percentuale di chi frequenta ore di religione  
PROVINCIA                                                             
VIBO VALENTIA                                                  97.48  
BARLETTA-ANDRIA-TRANI                                          97.48  
CROTONE                                                        97.45  
CAMPOBASSO                                                     97.22  
POTENZA                                                        97.10  
FOGGIA                                                         97.09  
ORISTANO                                                       96.70  
BRINDISI                                                       96.34  
CATANZARO                                                      96.08  
ISERNIA                                                        96.06  
ENNA                                                           95.25  
TARANTO                                                        95.25  
CALTANISSETTA                                                  95.25  
TRAPANI                       

In [65]:
data_2021.groupby('PROVINCIA').mean().sort_values(by='percentuale di chi frequenta ore di religione', ascending = True)[0:20]

alunni che frequentano  ANNOSCOLASTICO  totale alunni  \
PROVINCIA                                                              
BOLOGNA                        189.49       202021.00         284.41   
FIRENZE                        164.75       202021.00         270.21   
GORIZIA                        121.86       202021.00         170.57   
TRIESTE                        138.79       202021.00         205.71   
PRATO                          241.55       202021.00         339.82   
MODENA                         233.95       202021.00         315.66   
LIVORNO                        202.88       202021.00         277.70   
MANTOVA                        162.35       202021.00         218.81   
GROSSETO                       115.70       202021.00         152.12   
TORINO                         180.18       202021.00         244.50   
REGGIO EMILIA                  195.91       202021.00         257.07   
BRESCIA                        170.05       202021.00         224.98   
MILANO                         237.23       202021.00         312.78   
IMPERIA                        122.51       202021.00         158.59   
BIELLA                         105.27       202021.00         146.10   
RAVENNA                        239.81       202021.00         318.59   
FERRARA                        182.32       202021.00         239.85   
PIACENZA                       170.10       202021.00         223.62   
BERGAMO                        166.86       202021.00         215.55   
FERMO                          140.21       202021.00         163.51   

               alunni che non frequentano  \
PROVINCIA                                   
BOLOGNA                             94.92   
FIRENZE                            105.47   
GORIZIA                             48.71   
TRIESTE                             66.93   
PRATO                               98.27   
MODENA                              81.71   
LIVORNO                             74.82   
MANTOVA                             56.47   
GROSSETO                            36.42   
TORINO                              64.32   
REGGIO EMILIA                       61.17   
BRESCIA                             54.93   
MILANO                              75.55   
IMPERIA                             36.08   
BIELLA                              40.82   
RAVENNA                             78.78   
FERRARA                             57.53   
PIACENZA                            53.52   
BERGAMO                             48.69   
FERMO                               23.31   

               percentuale di chi frequenta ore di religione  
PROVINCIA                                                     
BOLOGNA                                                72.19  
FIRENZE                                                72.25  
GORIZIA                                                74.25  
TRIESTE                                                74.45  
PRATO                                                  75.54  
MODENA                                                 76.19  
LIVORNO                                                76.22  
MANTOVA                                                77.41  
GROSSETO                                               78.22  
TORINO                                                 78.93  
REGGIO EMILIA                                          79.10  
BRESCIA                                                79.15  
MILANO                                                 79.19  
IMPERIA                                                79.38  
BIELLA                                                 79.73  
RAVENNA                                                79.75  
FERRARA                                                79.77  
PIACENZA                                               80.19  
BERGAMO                                                80.21  
FERMO                                                  80.68

In [66]:
data_2021.groupby('DESCRIZIONETIPOLOGIAGRADOISTRUZIONESCUOLA').mean().sort_values(by='percentuale di chi frequenta ore di religione', ascending = False)

alunni che frequentano  \
DESCRIZIONETIPOLOGIAGRADOISTRUZIONESCUOLA                                   
IST PROF INDUSTRIA E ARTIGIANATO PER CIECHI                        213.00   
IST TECNICO ECONOMICO E TECNOLOGICO                                715.50   
SCUOLA PRIMARIA NON STATALE                                        110.78   
IST PROF ALBERGHIERO                                               221.62   
SCUOLA SEC. PRIMO GRADO NON STATALE                                102.15   
SCUOLA PRIMARIA                                                    143.47   
SCUOLA PRIMO GRADO                                                 191.82   
IST PROF PER L'AGRICOLTURA E L'AMBIENTE                            146.12   
LICEO CLASSICO                                                     508.14   
LICEO SCIENTIFICO                                                  555.75   
ISTITUTO TECNICO NAUTICO                                           352.26   
ISTITUTO TECNICO AGRARIO                                           229.18   
ISTITUTO MAGISTRALE                                                575.35   
IST TEC COMMERCIALE E PER GEOMETRI                                 346.84   
IST PROF PER I SERVIZI SOCIALI                                     257.81   
IST PROF PER I SERVIZI ALBERGHIERI E RISTORAZIONE                  400.38   
IST PROF PER L'AGRICOLTURA                                         131.56   
IST PROF INDUSTRIA E ATTIVITA' MARINARE                            145.38   
ISTITUTO TECNICO PER ATTIVITA' SOCIALI (GIA' ITF)                  379.17   
ISTITUTO D'ARTE                                                    281.52   
SCUOLA MAGISTRALE                                                  337.75   
ISTITUTO TECNICO COMMERCIALE                                       294.70   
ISTITUTO TECNICO PER GEOMETRI                                      173.16   
ISTITUTO TECNICO INDUSTRIALE                                       457.24   
LICEO ARTISTICO                                                    306.54   
IST PROF INDUSTRIA E ARTIGIANATO                                   185.62   
ISTITUTO TECNICO PER IL TURISMO                                    245.03   
IST PROF PER I SERVIZI COMMERCIALI                                 213.43   
IST PROF PER I SERVIZI COMMERCIALI E TURISTICI                     241.69   
ISTITUTO TECNICO AERONAUTICO                                       509.00   
IST PROF PER I SERVIZI TURISTICI                                   305.00   
IST PROF PER I SERVIZI COMM TUR E DELLA PUBB                       104.00   
IST PROF PER I SERVIZI PUBBLICITARI                                277.50   
IST PROF CINEMATOGRAFIA E TELEVISIONE                              577.00   
SCUOLA SEC. SECONDO GRADO NON STATALE                               44.13   
IST PROF INDUSTRIA E ARTIGIANATO PER SORDOMUTI                      60.50   

                                                   ANNOSCOLASTICO  \
DESCRIZIONETIPOLOGIAGRADOISTRUZIONESCUOLA                           
IST PROF INDUSTRIA E ARTIGIANATO PER CIECHI             202021.00   
IST TECNICO ECONOMICO E TECNOLOGICO                     202021.00   
SCUOLA PRIMARIA NON STATALE                             202021.00   
IST PROF ALBERGHIERO                                    202021.00   
SCUOLA SEC. PRIMO GRADO NON STATALE                     202021.00   
SCUOLA PRIMARIA                                         202021.00   
SCUOLA PRIMO GRADO                                      202021.00   
IST PROF PER L'AGRICOLTURA E L'AMBIENTE                 202021.00   
LICEO CLASSICO                                          202021.00   
LICEO SCIENTIFICO                                       202021.00   
ISTITUTO TECNICO NAUTICO                                202021.00   
ISTITUTO TECNICO AGRARIO                                202021.00   
ISTITUTO MAGISTRALE                                     202021.00   
IST TEC COMMERCIALE E PER GEOMETRI                      202021.00   
IST PROF PER I SERVI

## Considerazioni

* La **mancanza di dati** nel numero totale di studenti rende imprecisa questa analisi, è da notare come i due dataset principali (*studenti che frequentano le ore di religione per scuola* e *numero totale di studenti per scuola*) siano molto diversi come ordine di grandezza, e questo impedisce una fotografia adeguata del territorio nazionale.

* Tralasciando le scuole che non comunicano il loro numero di studenti, emergono comunque dei pattern interessanti: le **province e le regioni del Sud** registrano la partecipazione più alta, al contrario di quelle del **Nord** e del **Centro-Nord** (Toscana, Emilia-Romagna), le **scuole primarie e di primo grado**, assieme ai **licei**, vedono la partecipazione più alta alle ore di religione cattolica, al contrario degli indirizzi **tecnici e professionali**.

* è necessario far notare al Ministero la discrepanza tra il dataset sul *numero totale di studenti per scuola* (che rileva 31836 scuole) e quello sugli *studenti che frequentano le ore di religione per scuola* (che invece conta 52688 scuole).